src/modelmanager.cpp

```diff
@@ -757,6 +757,20 @@ Status ModelManager::loadConfig(const std::string& jsonFilename) {
 
     // Reading metric config only once per server start
     if (!this->metricConfigLoadedOnce) {
```

Comment:
It is not maintainable to edit loaded config.json into memory in the before applying the configuration. This can lead to hard investigate bugs when rapidjson object has different shape than file located on disk.

To address this issue, we could use existing `this->metricConfig.loadFromCLIString(config.metricsEnabled(), config.metricsList());` calls to modify `ModelManager` before loading config.json at all. Then, we can adjust applying the configuration to error in case metric config is already set. This way we can leave `ModelManager::startFromFile` and `ModelManager::loadConfig` without additional parameters such as `overrideMetricsEnable`.